external/loaders/loaders/mappers/_fine_res.py

```diff
@@ -0,0 +1,236 @@
+        return open_zarr(url, consolidated=False)
+
+
+def standardize_coords(
+    ds: xr.Dataset, time_shift=-timedelta(minutes=7, seconds=30)
+) -> xr.Dataset:
+    ds_shifted = ds.assign(time=ds.time + time_shift)
+    return gfdl_to_standard(ds_shifted).drop("tile")
+
+
+def _open_fine_resolution_dataset(
+    fine_url: str, additional_dataset_urls: Optional[Sequence[str]]
+) -> FineResBudget:
+
+    fine = open_zarr_maybe_consolidated(fine_url)
+    fine_shifted = standardize_coords(fine)
+
+    if additional_dataset_urls is not None:
+        additional_datasets = []
+        for url in additional_dataset_urls:
+            additional_datasets.append(open_zarr_maybe_consolidated(url))
+        merged = xr.merge([fine_shifted, *additional_datasets], join="inner")
+        if "latitude" in merged:
+            merged["latitude"] = merged.latitude.isel(time=0)
+        if "longitude" in merged:
+            merged["longitude"] = merged.longitude.isel(time=0)
+    else:
+        merged = fine_shifted
+
+    return merged
+
+
+class Approach(Enum):
+    apparent_sources_only = 1
+    apparent_sources_plus_nudging_tendencies = 2
+    apparent_sources_plus_dynamics_differences = 3
+    apparent_sources_extend_lower = 4
+
+
+def _compute_budget(
+    merged: xr.Dataset, approach: Approach, include_temperature_nudging: bool
+) -> MLTendencies:
+
+    merged["Q1"], merged["Q2"] = compute_fine_res_sources(
+        merged, include_temperature_nudging
+    )
+
+    if approach == Approach.apparent_sources_plus_nudging_tendencies:
+        merged["Q1"], merged["Q2"] = _add_nudging_tendencies(merged)
+    elif approach == Approach.apparent_sources_plus_dynamics_differences:
+        merged["Q1"], merged["Q2"] = _add_dynamics_differences(merged)
+    elif approach == Approach.apparent_sources_extend_lower:
+        merged["Q1"] = _extend_lower(merged["Q1"])
+        merged["Q2"] = _extend_lower(merged["Q2"])
+
+    return _ml_standard_names(merged)
+
+
+def _add_nudging_tendencies(merged: xr.Dataset):
+    with xr.set_options(keep_attrs=True):
+        Q1 = merged.Q1 + merged.air_temperature_tendency_due_to_nudging
+        Q2 = merged.Q2 + merged.specific_humidity_tendency_due_to_nudging
+    Q1.attrs.update(
+        {
+            "long_name": merged.Q1.attrs.get("long_name")
+            + " plus dynamics nudging tendencies",
+            "description": merged.Q1.attrs.get("description")
+            + " + dynamics nudging tendencies",
+        }
+    )
+    Q2.attrs.update(
+        {
+            "long_name": merged.Q2.attrs.get("long_name")
+            + " plus dynamics nudging tendencies",
+            "description": merged.Q2.attrs.get("description")
+            + " + dynamics nudging tendencies",
+        }
+    )
+    return Q1, Q2
+
+
+def _add_dynamics_differences(merged: xr.Dataset):
+    with xr.set_options(keep_attrs=True):
+        Q1 = merged.Q1 + merged.fine_minus_coarse_tendency_of_air_temperature_due_to_dynamics
```

Comment:
This is what Jeremy and I came up with about what this will likely be called. It's a mouthful but pretty unambiguous and vaguely CF-compliant.